libraries/Microsoft.Bot.Builder.Ai/LocaleConverterMiddleware.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.Bot.Builder.Middleware;
+using Microsoft.Bot.Schema;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Ai
+{
+    /// <summary>
+    /// Middleware to convert messages between different locales specified
+    /// </summary>
+    public class LocaleConverterMiddleware : IReceiveActivity
+    {
+        private LocaleConverter localeConverter;
+        private string fromLocale;
+        private string toLocale;
+
+        public LocaleConverterMiddleware(string fromLocale, string toLocale)
+        {
+            localeConverter = new LocaleConverter();
+            this.fromLocale = fromLocale;
+            this.toLocale = toLocale;
+        }
+
+        public async Task ReceiveActivity(IBotContext context, MiddlewareSet.NextDelegate next)
+        {
+            IMessageActivity message = context.Request.AsMessageActivity();
+            if (message != null)
+            {
+                if (!String.IsNullOrWhiteSpace(message.Text))
+                {
+                    await ConvertLocaleMessageAsync(message);
+                }
+            }
+            await next().ConfigureAwait(false);
```

Comment:
>await next().ConfigureAwait(false) [](start = 12, length = 34)

we're only translating incoming messages?  are we planning to support outgoing translations as well?